File: Source/JapaneseLanguageTools.Contracts/Models/Requests/SubmitCompletedCharacterExerciseDetailsRequest.Model.cs
using System.Collections.Generic;

using JapaneseLanguageTools.Contracts.Models.Responses;

namespace JapaneseLanguageTools.Contracts.Models.Requests;

public class SubmitCompletedCharacterExerciseDetailsRequestModel
{
    public IList<CharacterExerciseModel> CompletedItems { get; set; } = [];

    public required GenerateCharacterExerciseBatchResponseModel OriginalResponse { get; set; }
}

File: Source/JapaneseLanguageTools.ReactSpa/source/models/requests/SubmitCompletedCharacterExerciseDetailsRequest.Model.ts
import CharacterExerciseModel from "@/models/CharacterExercise.Model";
import GenerateCharacterExerciseBatchResponseModel from "@/models/responses/GenerateCharacterExerciseBatchResponse.Model";

export default interface SubmitCompletedCharacterExerciseDetailsRequestModel {
  completedItems: CharacterExerciseModel[];

  originalResponse: GenerateCharacterExerciseBatchResponseModel;
}
